redfish-core/include/license_messages.hpp

```diff
@@ -0,0 +1,129 @@
+        error["message"] = "A general error has occurred. See Resolution for "
+                           "information on how to resolve the error.";
+    }
+
+    // This check could technically be done in in the default construction
+    // branch above, but because we need the pointer to the extended info field
+    // anyway, it's more efficient to do it here.
+    auto& extendedInfo = error[messages::messageAnnotation];
+    if (!extendedInfo.is_array())
+    {
+        extendedInfo = nlohmann::json::array();
+    }
+
+    extendedInfo.push_back(message);
+}
+
+inline nlohmann::json licenseInstalled(const std::string& arg1)
+{
+    return nlohmann::json{{"@odata.type", "#Message.v1_0_0.Message"},
+                          {"MessageId", "License.1.0.0.LicenseInstalled"},
+                          {"Message", "The license has been installed."},
+                          {"MessageArgs", {arg1}},
+                          {"Severity", "OK"},
+                          {"Resolution", "None."}};
+}
+
+void licenseInstalled(crow::Response& res)
+{
+    res.result(boost::beast::http::status::ok);
+    addMessageToErrorJson(res.jsonValue, licenseInstalled("LicenseInstalled"));
+}
+
+inline nlohmann::json invalidLicense()
+{
+    return nlohmann::json{{"@odata.type", "#Message.v1_0_0.Message"},
+                          {"MessageId", "License.1.0.0.InvalidLicense"},
+                          {"Message",
+                           "The content of the license was not recognized, is "
+                           "corrupted, or is invalid."},
+                          {"MessageArgs", {}},
+                          {"Severity", "Critical"},
+                          {"Resolution", "None."}};
+}
+
+void invalidLicense(crow::Response& res)
+{
+    res.result(boost::beast::http::status::bad_request);
+    addMessageToErrorJson(res.jsonValue, invalidLicense());
+}
+
+inline nlohmann::json installFailed(const std::string& arg1)
+{
+    return nlohmann::json{
+        {"@odata.type", "#Message.v1_0_0.Message"},
+        {"MessageId", "License.1.0.0.InstallFailed"},
+        {"Message", "Failed to install the license.  Reason: %1."},
+        {"MessageArgs", {arg1}},
+        {"Severity", "Critical"},
+        {"Resolution", "None."}};
+}
+
+void installFailed(crow::Response& res, const std::string& reason)
+{
+    res.result(boost::beast::http::status::internal_server_error);
+    addMessageToErrorJson(res.jsonValue, installFailed(reason));
+}
+
+inline nlohmann::json notApplicableToTarget()
+{
+    return nlohmann::json{
+        {"@odata.type", "#Message.v1_0_0.Message"},
+        {"MessageId", "License.1.0.0.notApplicableToTarget"},
+        {"Message",
+         "Indicates that the license is not applicable to the target."},
+        {"MessageArgs", {}},
+        {"Severity", "Critical"},
+        {"Resolution", "None."}};
+}
+
+void notApplicableToTarget(crow::Response& res)
+{
+    res.result(boost::beast::http::status::internal_server_error);
```

Comment:
This can be a 400 - Bad Request. Upon seeing this response, client should change the license data and send a fresh request again

Reply:
done